modules/porous_flow/src/base/PorousFlowApp.C

```diff
@@ -66,13 +66,15 @@ associateSyntaxInner(Syntax & syntax, ActionFactory & /*action_factory*/)
 
   registerMooseObjectTask("add_porous_flow_bc", PorousFlowSinkBC, false);
   addTaskDependency("add_porous_flow_bc", "add_bc");
+  addTaskDependency("resolve_optional_materials", "add_porous_flow_bc");
 
   // Task dependency and syntax for action to automatically add PorousFlow materials
   registerSyntax("PorousFlowAddMaterialAction", "Materials");
 
   // Task dependency and syntax for action to automatically add PorousFlowJoiner materials
   registerTask("add_joiners", /*is_required=*/false);
   addTaskDependency("add_joiners", "add_material");
+  addTaskDependency("add_master_action_material", "add_joiners");
```

Comment:
how did you determine this whole list of extra dependencies?
Was this just to make the test pass or do you have a list that you impose on each module?

Reply:
See previous comment. In general the new task would get inserted immediately behind the prereq task in the task list given in `Moose.C`. If you look in `Moose.C`, `add_master_action_material` is the task that occurs right after `add_material`